### PR TITLE
fixup: tidy code styles & clarify var names

### PR DESCRIPTION
**Pull request by GeoffreyBooth**
Implements my notes from https://github.com/nodejs/node/pull/42623. I put them in separate commits so that you can cherry-pick if you agree with some and not others. Happy to discuss.

### CODE REVIEW

File: lib/internal/modules/esm/loader.js [hunk 1]
```diff
@@ -468,7 +468,7 @@ class ESMLoader {
    */
   async import(specifiers, parentURL, importAssertions) {
     const wasArr = ArrayIsArray(specifiers);
-    if (!wasArr) specifiers = [specifiers];
+    if (!wasArr) { specifiers = [specifiers]; }
```
Comment (JakobJingleheimer) [inline lib/internal/modules/esm/loader.js:471]: oow, I like it better with the curly brackets and it still a 1-liner 😀

File: lib/internal/modules/esm/loader.js [hunk 2]
```diff
@@ -507,20 +507,23 @@ class ESMLoader {
...
       fn: loader,
       url: loaderFilePath,
-    } = loaders[i];
-    let chainFinished = i === 0;
+    } = loaders[hookIndex];
+    let chainFinished = hookIndex === 0;
+
+    const next = (nextUrl, ctx = context) => {
+      // `next` has been called, so decrement our pointer
+      --hookIndex;
```
Comment (JakobJingleheimer) [inline lib/internal/modules/esm/loader.js:519]: actually, maybe yes this is better separated for someone who doesn't know the subtle difference between `--hookIndex` and `hookIndex--`

**Review by JakobJingleheimer — APPROVED** [review]
🙌

File: lib/internal/modules/esm/loader.js (hunk 2 above)
Comment (GeoffreyBooth) [reply lib/internal/modules/esm/loader.js:519]: And when it’s on its own line, the different doesn’t matter. You could rewrite this as `hookIndex--;` if that seems clearer.

Comment (JakobJingleheimer) [reply lib/internal/modules/esm/loader.js:519]: true. too late—you missed the ship 😜

**Pull request merged**